### PR TITLE
[SYCL][CUDA] Test case covers unnamed lambda and hierarchical kernels

### PR DESCRIPTION
**Pull request by JackAKirk**
Corrected header documentation. All tests pass using CUDA.
Signed-off-by: JackAKirk <jack.kirk@codeplay.com>

### CODE REVIEW

File: SYCL/InorderQueue/in_order_kernels.cpp [hunk 1]
```diff
@@ -1,13 +1,12 @@
...
-// CUDA does not support unnamed lambdas.
 //
 // RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple -fsycl-unnamed-lambda %s -o %t.out
 // RUN: %HOST_RUN_PLACEHOLDER %t.out
 // RUN: %ACC_RUN_PLACEHOLDER %t.out
 // RUN: %CPU_RUN_PLACEHOLDER %t.out
 // RUN: %GPU_RUN_PLACEHOLDER %t.out
 
-//==------ oq_kernels.cpp - SYCL ordered queue kernel shortcut test --------==//
+//==------ InorderQueue.cpp - SYCL ordered queue kernel shortcut test
```
Comment (v-klochkov) [inline SYCL/InorderQueue/in_order_kernels.cpp]: Minor comment: why "InorderQueue.cpp"? The test name is in_order_kernels.cpp.

The comment could be without the test name at all, something like: // This test checks the kernel call shortcut methods of SYCL ordered queue" or "SYCL ordered queue kernel shortcut test"
```suggestion
// SYCL ordered queue kernel shortcut test
```

File: SYCL/InorderQueue/in_order_kernels.cpp [hunk 2]
```diff
@@ -42,28 +41,62 @@ int main() {
...
+    });
+
     q.parallel_for<class NDFoo>(NDR, [=](nd_item<1> Item) {
       auto i = Item.get_global_id(0);
       A[i]++;
     });
 
+    q.submit([&](handler &cgh) {
+      cgh.parallel_for_work_group<class WkGrp>(
+          range<1>{4}, range<1>{2}, [=](group<1> myGroup) {
```
Comment (v-klochkov) [inline SYCL/InorderQueue/in_order_kernels.cpp]: minor/nfc thing:
```suggestion
          range<1>{N / 2}, range<1>{2}, [=](group<1> myGroup) {
```

File: SYCL/InorderQueue/in_order_kernels.cpp [hunk 3]
```diff
@@ -42,28 +41,62 @@ int main() {
...
+            auto j = myGroup.get_id(0);
+            myGroup.parallel_for_work_item(
+                [&](h_item<1> it) { A[(j * 2) + it.get_local_id(0)]++; });
+          });
+    });
+
+    q.submit([&](handler &cgh) {
+      cgh.parallel_for_work_group(
+          range<1>{4}, range<1>{2}, [=](group<1> myGroup) {
```
Comment (v-klochkov) [inline SYCL/InorderQueue/in_order_kernels.cpp]:
```suggestion
          range<1>{N / 2}, range<1>{2}, [=](group<1> myGroup) {
```

File: SYCL/InorderQueue/in_order_kernels.cpp (hunk 1 above)
Comment (JackAKirk) [reply SYCL/InorderQueue/in_order_kernels.cpp]: Yes that was a mistake it should have read 'in_order_kernels.cpp'.  I'll switch it to your suggestion. Thanks

File: SYCL/InorderQueue/in_order_kernels.cpp (hunk 2 above)
Comment (JackAKirk) [reply SYCL/InorderQueue/in_order_kernels.cpp]: Thanks, I'll change this.

**Review by v-klochkov — APPROVED** [review]

**Pull request merged**